FoodFriends/app/src/main/java/com/example/foodfriends/misc/DistanceRestaurantServer.java

```diff
@@ -0,0 +1,142 @@
+package com.example.foodfriends.misc;
+
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+
+import com.example.foodfriends.models.Friends;
+import com.example.foodfriends.models.Restaurant;
+import com.example.foodfriends.models.UserLike;
+import com.example.foodfriends.models.UserToGo;
+import com.example.foodfriends.observable_models.RestaurantObservable;
+import com.example.foodfriends.observable_models.UserObservable;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class DistanceRestaurantServer {
+    public static final String YELP_URL = "https://api.yelp.com/v3/businesses/search";
+    private static final String TAG = "Restaurant Server";
+    private int maxRadiuskm = 40; //in kilometers
+    private int offset;
+    private List<RestaurantObservable> observedRestaurants;
+    private UserObservable user;
+
+
+
+    /**
+     * Constructor
+     */
+    public DistanceRestaurantServer(List<RestaurantObservable> restaurantList) {
+        this.offset = 0;
+        this.observedRestaurants = restaurantList;
+        this.user = new UserObservable(ParseUser.getCurrentUser());
+    }
+
+
+    /**
+     * Resets the instance variables to allow for a new search without offsets and at Parse
+     */
+    public void reset() {
+        this.offset = 0;
+    }
+
+
+    public UserObservable getUser() {
+        return user;
+    }
+
+    public void setUser(UserObservable user) {
+        this.user = user;
+    }
+
+
+    /**
+     * Adds restaurant from parse likes, togos, or general or calls yelpquery to do so depending on source
+     */
+    public void findRestaurants(String apiKey, @Nullable RestaurantListener listener) {
+        yelpQuery(apiKey, listener);
+    }
+
+
+    /**
+     * Adds restaurants returned from Yelp Query to the list that the explore adapter uses
+     */
+    public void yelpQuery(String apiKey,  @Nullable RestaurantListener listener) {
```

Comment:
Should this be private? This looks like a low level infra thing. 

Patterns that are usually used is, an internal and an external YelpQuery. external methods can call with the internal method with different parameters.